homeassistant/components/tasmota/binary_sensor.py

```diff
@@ -41,45 +56,52 @@ class TasmotaBinarySensor(
 ):
     """Representation a Tasmota binary sensor."""
 
-    def __init__(self, **kwds):
+    _tasmota_entity: tasmota_switch.TasmotaSwitch
+
+    def __init__(self, **kwds: Any) -> None:
         """Initialize the Tasmota binary sensor."""
-        self._delay_listener = None
-        self._state = None
+        self._delay_listener: Callable | None = None
+        self._on_off_state: bool | None = None
 
         super().__init__(
             **kwds,
         )
 
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to MQTT events."""
+        self._tasmota_entity.set_on_state_callback(self.on_off_state_updated)
+        await super().async_added_to_hass()
+
     @callback
-    def off_delay_listener(self, now):
+    def off_delay_listener(self, now: Any) -> None:
```

Comment:
I believe it passes a datetime in here.

```suggestion
    def off_delay_listener(self, now: datetime) -> None:
```

Reply:
You're right, fixed in 0e81f3a

It would be nice if `async_call_later` was better typed than this though:
```py
def async_call_later(
    hass: HomeAssistant,
    delay: float | timedelta,
    action: HassJob | Callable[..., Awaitable[None] | None],
) -> CALLBACK_TYPE:
```